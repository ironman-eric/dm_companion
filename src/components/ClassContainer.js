import AttributePicker from './AttributePicker';
import dndClass from '../data/dndClass';

function ClassContainer({ onChangeValue }) {
  return (
    <div className="mb-4">
      <label className="block text-gray-700 text-sm font-bold mb-2">
        Select a Class
      </label>
      <AttributePicker name="dndClass" items={Object.keys(dndClass)} onOptionChange={onChangeValue} />      
    </div>
  );
}

export default ClassContainer;
