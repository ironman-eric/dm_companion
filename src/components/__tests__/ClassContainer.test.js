import React from "react";
import renderer from "react-test-renderer";
import ClassContainer from "../ClassContainer";

describe("ClassContainer", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<ClassContainer />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
