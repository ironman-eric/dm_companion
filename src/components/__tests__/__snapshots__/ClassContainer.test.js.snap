// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassContainer Matches DOM Snapshot 1`] = `
<div
  className="mb-4"
>
  <label
    className="block text-gray-700 text-sm font-bold mb-2"
  >
    Select a Class
  </label>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Barbarian"
    />
      
    Barbarian
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Bard"
    />
      
    Bard
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Cleric"
    />
      
    Cleric
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Fighter"
    />
      
    Fighter
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Monk"
    />
      
    Monk
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Ranger"
    />
      
    Ranger
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Rogue"
    />
      
    Rogue
  </span>
  <span
    className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
  >
    <input
      name="dndClass"
      type="radio"
      value="Wizard"
    />
      
    Wizard
  </span>
</div>
`;
